lib/puppet/parser/functions/new.rb

```diff
@@ -864,7 +866,7 @@
 | b | The data is in base64 encoding, padding as required by base64 strict is added by default
 | u | The data is in URL safe base64 encoding
 | B | The data is in base64 strict encoding
-| s | The data is a puppet string used verbatim as a binary byte sequence
+| s | The data is a puppet string seen as a binary sequence of bytes with ignored encoding
```

Comment:
Eh, it works the other way around. Ruby encodes when an operation takes place on the string. It is always a binary byte sequence internally. The ruby method that returns the bytes simply returns them. If you implemented this in Java, the Binary byte buffer created from a UTF-16 string would contain two bytes per character (or more since UTF-16 also has multi-chars IIRC). Do you have a suggestion on how to phrase this? Technically it takes a copy and enforces an '8 bit ascii' encoding (which is ruby's "no encoding, any byte sequence allowed" - i.e. "ignored encoding").